common/src/test/java/com/revenuecat/purchases/common/OfferingsTest.kt

```diff
@@ -23,67 +23,92 @@ import org.robolectric.annotation.Config
+    private val monthlyDuration = "P1M"
+    private val annualDuration = "P1Y"
 
     @Test
     fun `Package is not created if there are no valid products`() {
+        val storeProductAnnual = getStoreProduct(productIdentifier, annualDuration, annualBasePlan)
+        val products = mapOf(productIdentifier to listOf(storeProductAnnual))
+
         val packageToTest = getPackageJSON().createPackage(
-            getProductsByGroupId("com.myproduct.annual" to "P1Y"),
+            products,
             "offering"
         )
         assertThat(packageToTest).isNull()
     }
 
     @Test
     fun `Package is created if there are valid products`() {
-        val products: Map<String, List<StoreProduct>> = getProductsByGroupId()
-        val packageJSON = getPackageJSON(
+        val storeProductMonthly = getStoreProduct(productIdentifier, monthlyDuration, monthlyBasePlan)
+        val storeProductAnnual = getStoreProduct(productIdentifier, annualDuration, annualBasePlan)
+
+        val products = mapOf(productIdentifier to listOf(storeProductMonthly, storeProductAnnual))
+
+        val monthlyPackageJSON = getPackageJSON(
             packageIdentifier = PackageType.MONTHLY.identifier!!,
-            productIdentifier = monthlyProductIdentifier,
-            productGroupIdentifier = monthlyProductGroupIdentifier
+            productIdentifier = productIdentifier,
+            basePlanId = monthlyBasePlan
         )
-        val packageToTest = packageJSON.createPackage(products, "offering")
+
+        val packageToTest = monthlyPackageJSON.createPackage(products, "offering")
         assertThat(packageToTest).isNotNull
-        assertThat(packageToTest!!.product).isEqualTo(products[monthlyProductGroupIdentifier]?.get(0))
+        assertThat(packageToTest!!.product).isEqualTo(products[productIdentifier]?.get(0))
         assertThat(packageToTest.identifier).isEqualTo(PackageType.MONTHLY.identifier)
         assertThat(packageToTest.packageType).isEqualTo(PackageType.MONTHLY)
     }
 
     @Test
     fun `Package is created for migrated products with same product identifier and group`() {
-        val productIdPreBC5 = "onemonth_freetrial"
-        val products: Map<String, List<StoreProduct>> = getProductsByGroupId(productIdPreBC5 to "P1M")
-        val packageJSON = getPackageJSON(
-            packageIdentifier = PackageType.MONTHLY.identifier!!,
-            productIdentifier = productIdPreBC5,
-            productGroupIdentifier = productIdPreBC5
-        )
-        val packageToTest = packageJSON.createPackage(products, "offering")
-        assertThat(packageToTest).isNotNull
-        assertThat(packageToTest!!.product).isEqualTo(products[productIdPreBC5]?.get(0))
-        assertThat(packageToTest.identifier).isEqualTo(PackageType.MONTHLY.identifier)
-        assertThat(packageToTest.packageType).isEqualTo(PackageType.MONTHLY)
+        // TODO BC5: not sure how migrated products will look like
```

Comment:
Left this as a question because I am not sure how the JSON looks for products migrated from BC4 to BC5. cc @swehner @dalshamaa

Reply:
The idea is that all BC5 clients will receive productId and product_plan_id for each package that is mapped to a subscritpion product. There shouldn't be a difference whether the product is the BW-compatible one or not.